source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -54,13 +65,49 @@ bool isTargetClusterLambdaGateway(Upstream::ClusterInfo const& cluster_info) {
   return egress_gateway_it->second.bool_value();
 }
 
+bool isContentTypeTextual(const Http::RequestOrResponseHeaderMap& headers) {
+  // If transfer-encoding is anything other than 'identity' (i.e. chunked, compress, deflate or
+  // gzip) then we want to base64-encode it regardless of the content-type value.
+  if (headers.TransferEncoding()) {
+    std::string encoding{headers.TransferEncoding()->value().getStringView()};
+    if (Http::LowerCaseString(encoding).get() != "identity") {
+      return false;
+    }
+  }
+
+  // If we don't know the content-type, then we can't make any assumptions.
+  if (!headers.ContentType()) {
+    return false;
+  }
+
+  const Http::LowerCaseString content_type_value{
+      std::string(headers.ContentType()->value().getStringView())};
+  if (content_type_value.get() == "application/json") {
+    return true;
+  }
+
+  if (content_type_value.get() == "application/javascript") {
+    return true;
+  }
+
+  if (content_type_value.get() == "application/xml") {
+    return true;
+  }
+
+  if (absl::StartsWith(content_type_value.get(), "text/")) {
+    return true;
+  }
+
+  return false;
```

Comment:
I have it that way to show which branches are covered in the coverage report. It also makes stepping through the code easier when debugging because you can see which branch evaluated to true.

Reply:
Okay, cool. This was kind of a nit anyway.